internal/statesync/metrics.go

```diff
@@ -0,0 +1,91 @@
+package statesync
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/discard"
+	"github.com/go-kit/kit/metrics/prometheus"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// MetricsSubsystem is a subsystem shared by all metrics exposed by this package.
+	MetricsSubsystem = "statesync"
+)
+
+// Metrics contains metrics exposed by this package.
+type Metrics struct {
+	TotalSnapshots     metrics.Counter
+	ChunkProcess       metrics.Gauge
```

Comment:
It's not clear what this field denotes—is this a number of chunks that have been processed? That seems obvious, but the fact that it's a gauge rather than a counter made me wonder. (If so, maybe consider `ChunksProcessed`).

Reply:
rename to `ChunkProcessAvgTime`